### PR TITLE
feat(lsp): report fswatch errors

### LINKED ISSUE #27713: No error notifications from `fswatch`
### Problem

I tried to check if and how `fswatch` works for me - in any way at first but couldn't find any such process. It turned out that it is started for phpactor LS so I got [the command](https://github.com/neovim/neovim/blob/dbf6be296df6da28151afdcb96ceb8ba2076cc65/runtime/lua/vim/_watch.lua#L274-L284) into one-liner and ran it directly from console. I saw the stderr there (it was `Event queue overflow`) but not in nvim. 

### Steps to reproduce

On Linux try to (for example) decrease `/proc/sys/fs/inotify/max_queued_events` enough to get _event queue overflow_ error and see if nvim warns about it. 

### Expected behavior

Notify about errors using `stderr` callback to `vim.system` or in another way.

### Neovim version (nvim -v)

v0.10.0-dev-2500+gdc8c086c7

### Vim (not Nvim) behaves the same?

-

### Operating system/version

Debian Sid

### Terminal name/version

alacritty

### $TERM environment variable

alacritty

### Installation

from repo

### PR DESCRIPTION
**Pull request by lewis6991**
Resolves #27713

### CODE REVIEW

File: runtime/lua/vim/_watch.lua [hunk 1]
```diff
@@ -283,6 +283,13 @@ function M.fswatch(path, opts, callback)
     '/.git/',
     path,
   }, {
+    stderr = function(err, data)
+      if err then
+        error(err)
+      end
+
+      vim.notify(data, vim.log.levels.ERROR)
```
Comment (przepompownia) [inline runtime/lua/vim/_watch.lua:291]:
```suggestion
      if data and #vim.trim(data) > 0 then
        vim.notify(data, vim.log.levels.ERROR)
      end
```
For me it prevents from displaying empty notification.

**Comment by przepompownia** [comment]
After the change empty stderr entries are skipped from notification for me.

File: runtime/lua/vim/_watch.lua [hunk 2]
```diff
@@ -283,6 +283,15 @@ function M.fswatch(path, opts, callback)
     '/.git/',
     path,
   }, {
+    stderr = function(err, data)
+      if err then
+        error(err)
+      end
+
+      if data and #vim.trim(data) > 0 then
+        vim.notify(data, vim.log.levels.ERROR)
```
Comment (przepompownia) [inline runtime/lua/vim/_watch.lua:292]:
```suggestion
        vim.schedule(function ()
          vim.notify(data, vim.log.levels.ERROR, {title = 'LSP Watch'})
        end)
```

Comment (przepompownia) [reply runtime/lua/vim/_watch.lua:292]: because of `vim/_editor.lua:0: E5560: nvim_err_writeln must not be called in a lua loop callback`.

Comment (przepompownia) [reply runtime/lua/vim/_watch.lua:292]: I initially tested it with Noice I initially tested it with Noice, so there was no such error before.

**Comment by przepompownia** [comment]
Is there anything that blocks it from merging?

**Comment by lewis6991** [comment]
I want to tweak the message so it's obvious it's coming from fswatch

**Pull request merged**